src/ConnectedMode.UnitTests/Binding/ImportsBeforeFileGeneratorTests.cs

```diff
@@ -0,0 +1,103 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.IO;
+using System.IO.Abstractions;
+using SonarLint.VisualStudio.ConnectedMode.Binding;
+
+namespace SonarLint.VisualStudio.ConnectedMode.UnitTests.Binding
+{
+    [TestClass]
+    public class ImportsBeforeFileGeneratorTests
+    {
+        [TestMethod]
+        public void FileDoesNotExist_CreatesFile()
+        {
+            string pathToDirectory = GetPathToImportBefore();
+            string pathToFile = Path.Combine(pathToDirectory, "SonarLint.targets");
+            string expectedResult = "<Project xmlns=\"http://schemas.microsoft.com/developer/msbuild/2003\">\r\n  <ItemGroup />\r\n</Project>";
```

Comment:
Checking the exact file content each time makes the tests quite fragile, as we'll have to keep updating them as we change the content.

It might be simpler to have one test that checks that the content is a valid MSBuild project file i.e. it can be parsed as XML and starts with a `Project` element with the expected attribute.

The other tests could then just check that the file system method are called as expected, without explicitly checking the content.